### PR TITLE
MAINT: Simplify boolean expression

### PR DESCRIPTION
**Pull request by remyleone**

### CODE REVIEW

File: scipy/cluster/hierarchy.py [hunk 1]
```diff
@@ -2292,7 +2292,7 @@ def _dendrogram_calculate_info(Z, p, truncate_mode,
         # If the node is a leaf node but corresponds to a non-single cluster,
         # its label is either the empty string or the number of original
         # observations belonging to cluster i.
-        if i < 2*n - p and i >= n:
+        if 2 * n - p > i >= n:
```
Comment (rgommers) [inline scipy/cluster/hierarchy.py]: nitpick: the extra spaces here are undesired, ``2*n`` is more readable and PEP8 was specifically changed to allow that.

**Comment by rgommers** [comment]
useful cleanup, thanks @sieben

**Comment by rgommers** [comment]
What tool are you using to check for these things?

**Comment by sieben** [comment]
Pycharm :) inspect code feature

**Comment by rgommers** [comment]
not too bothered by the ``2 * n``, in it goes. Thanks @sieben

**Pull request merged**